week8/week8/ContentView.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  ContentView.swift
+//  Mod8JSON
+//
+//  Created by Nick Ryan on 6/8/24.
+//
+
+import SwiftUI
+
+struct ContentView: View {
+  @StateObject private var universityStore = UniversityStore()
+  
+  
+  var body: some View {
+    NavigationStack {
+      VStack {
+        if universityStore.downloadProgress < 1.0 {
+          ProgressView(value: universityStore.downloadProgress)
+            .padding()
+            .onAppear {
+              universityStore.loadData()
+            }
```

Comment:
If you change `loadData` to async, you can use `task` instead. And also move to NavigationStack.
```
NavigationStack {
  // rest of the views
}
.task {
  await universityStore.loadData()
}
```